docs/contribute/cumulative-docs/content-patterns.md

```diff
@@ -0,0 +1,383 @@
+---
+navigation_title: "Example scenarios"
+---
+
+# Example scenarios
+
+## Code block content varies [code-block]
+
+Often the content in a code block will vary between situations (versions, deployment types, etc).
+There are a couple possible solutions.
+
+### Solution A: Use a code callout [code-block-callout]
+
+Using a code callout is the lightest-touch solution, but might not be sufficient in all cases.
+
+**When to use a code callout**:
+
+* The code block and its callouts fit vertically on a typical laptop screen.
+  This will reduce the risk of users copying the code snippet without reading the information in the callout.
+* Syntax is either just added or just removed — syntax is not modified.
+  It is difficult to communicate that some syntax is needed in more than one situation but varies depending on the situation.
+* The code block will not require more than 3 `applies_to`-related callouts.
+  At that point, the code becomes more difficult to read and use.
+
+**Best practices**:
+
+* Place the badge at the beginning of the callout.
+
+**Example**: A new option was made available in 9.1.0.
+
+::::{image} ./images/example-code-block-callout.png
+:screenshot:
+:alt:
+::::
+
+### Solution B: Use tabs [code-block-tabs]
```

Comment:
I wonder if this should be something that we build into docs-builder. Tabs is a good workaround for now, but a custom component built for handling large variation in code examples over versions or deployment types could be better. Maybe related to https://github.com/elastic/docs-builder/issues/1436

Reply:
Hmmm, how would it look like? Something wrapping a block with a version menu? Horizontal tabs?

Reply:
This discussion should move to https://github.com/elastic/docs-builder/issues/1436.